docs/development.md

```diff
@@ -0,0 +1,264 @@
+<!-- 
```

Comment:
Thoughts on moving this to `CONTRIBUTING.md` so [github picks it up](https://help.github.com/en/articles/setting-guidelines-for-repository-contributors#adding-a-contributing-file)?

Reply:
I typically view the `CONTRIBUTING.md` file to be a bit higher level, linking to code of conducts, including a developer's certificate of origin, and linking to other docs necessary for contributing, such as a developer environment guide, style guides, etc. For an example, see [here](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md). On reflection, some overlap between them. I am fine with consolidating if you think it best.

Reply:
I am fine with merging this as is and iterating. I am fine with either.

Reply:
Added a contributing doc. In the abstract, I see "contributing" and "development" as two separate, but possible overlapping things. For example, I may want to locally develop, but never contribute, and I may want to contribute, but never develop. The contributing guide I included is based on the one I linked to above and includes a bit more hand holding for neophytes. Maybe this is unnecessary.

Reply:
That makes sense to me! Thank you.